engine/access/rpc/engine.go

```diff
@@ -298,24 +313,25 @@ func (e *Engine) serveUnsecureGRPC() {
 	e.addrLock.Lock()
 	e.unsecureGrpcAddress = l.Addr()
 	e.addrLock.Unlock()
-
 	e.log.Debug().Str("unsecure_grpc_address", e.unsecureGrpcAddress.String()).Msg("listening on port")
+	ready()
 
 	err = e.unsecureGrpcServer.Serve(l) // blocking call
 	if err != nil {
-		e.log.Fatal().Err(err).Msg("fatal error in unsecure grpc server")
+		e.log.Err(err).Msg("fatal error in unsecure grpc server")
+		ctx.Throw(err)
 	}
 }
 
-// serveSecureGRPC starts the secure gRPC server
-// When this function returns, the server is considered ready.
-func (e *Engine) serveSecureGRPC() {
-
+// serveSecureGRPCWorker is a worker routine which starts the secure gRPC server.
+// The ready callback is called after the server address is bound and set.
+func (e *Engine) serveSecureGRPCWorker(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
```

Comment:
serveUnsecureGRPCWorker and serveSecureGRPCWorker seem like they could be one function with some parameters

`func (e *Engine) serveGRPCWorker(name, address string, server *grpc.Server) func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc)`